### DISCUSSION TITLE
Gromacs Import Warning

### DISCUSSION THREAD

**Discussion by angeyx** [Becksteinlab/GromacsWrapper#230] [Q&A] [answered]
Hi, I installed gromacs wrapper after installing Gromacs. But when I tried `import gromacs` in jupyter lab, it gave me a warning

`c:\users\carno\appdata\local\programs\python\python38\lib\site-packages\gromacs\__init__.py:286: GromacsImportWarning: Some Gromacs commands were NOT found; maybe source GMXRC first? The following are missing:
['release']`
I also couldn't call any gromacs tools. When I open the .gromacswrapper.cfg, `release`, `gmrx`, and `tools` are all empty, which I don't think is normal.

Any ideas on how to solve this? I'm new to gromacs and using Win10 system & gromacs-2022.3. Happy to provide more information if needed.

**Comment by orbeckst** [comment]
The warning really says that GromacsWrapper could not find your `gmx` executable on your PATH. I don't have a windows machine so I can't test things directly.  Ultimately, this is a question about [configuration](https://gromacswrapper.readthedocs.io/en/latest/configuration.html) but before I can suggest anything specific, let me ask a few questions:

1. How did you install GromacsWrapper?
2. How did you install GROMACS?
3. Where is GROMACS installed?
4. Can you open a terminal (also known as command line or shell) and work there?
5. How do you launch Jupyter?

**Reply by angeyx** [reply]
1. I installed GromacsWrapper via both pip and conda install. The above error is when I launched it through the command prompt.  A similar error will happen if I launch Jupyter through the anaconda prompt.
2. I installed GROMACS via Ubuntu 
3. it was installed it in `c:\users\carno\`
4. If I directly run python and `import gromacs` in the command prompt, it will give the same error, if that is what you're asking.

**Reply by orbeckst** [reply]
How do you run gmx from the command line?

What is the full path of the `gmx` executable?

**Reply by angeyx** [reply]
I was trying to trouble-shoot GROMACS these few days. I seem to have two versions installed at two places. 

- one at `/usr/bin/gmx` , version `2020.10 Ubuntu-2020.1-1.`, (I think I installed accidentally following a tutorial, directly via Ubuntu `apt install gromacs`) . I can run this directly via Ubuntu.
- one at `usr/local/gromacs/bin/gmx`, version `gmx, 2022.3` (I downloaded the package from GROMACS and followed the instructions). I have to source it first before running.

I probably need to remove one of them and go with one.

**Comment by angeyx** [comment, marked as answer]
Hi an update -- I figured it out. I think the problem is the windows and Linux/WSL environments. I installed Anaconda for windows prior to I installed Ubuntu, and used it to install `gromacswrapper` and run my code. `gromacswrapper` has trouble finding my `gmx` because they reside in different file systems, so even when I modify the `GMXRC=...` in the `config` file, it can't find/access it. Now I installed Anaconda for Linux from Ubuntu, and then installed `gromacswrapper`, and it worked finally! Thanks for pointing out the PATH problem -- that points me in the right direction.

**Comment by orbeckst** [comment]
Glad that you figured it out!
